sale_order_import/tests/test_sale_order_import.py

```diff
@@ -0,0 +1,241 @@
+            self.partner_02.property_payment_term_id,
+            "Sale Order Payment term field data does not match"
+            " with Test Partner Payment term condtion",
+        )
+        self.assertEqual(
+            sale_order[0].picking_policy,
+            "direct",
+            'Sale order picking policy value does not match with "direct"',
+        )
+        self.assertEqual(
+            sale_order[0].team_id,
+            self.team,
+            'From test Sale Order file the "Team"' " does not match with test records",
+        )
+        self.assertEqual(
+            sale_order[0].carrier_id.id,
+            self.free_delivery,
+            'From test Sale Order file the "Carrier"'
+            " does not match with test records",
+        )
+        self.assertEqual(
+            sale_order[0].warehouse_id,
+            self.warehouse,
+            'From test Sale Order file the "warehouse" '
+            "does not match with test records",
+        )
+        self.assertEqual(
+            sale_order[0].note,
+            self.note,
+            "From test Sale Order file the customer does not match with records",
+        )
+
+    def test_update_error_log_02(self):
+        """
+            This test evaluates the import test sales order CSV
+             file and it update the error log records if any issue occurred while import
+         """
+
+        # Create Attachment to add as argument in `update_error_log`
+        ir_attachment = self.env["ir.attachment"].create(
+            {
+                "name": "test_sale_order.csv",
+                "datas": base64.encodestring(self.generated_file),
+                "datas_fname": "test_sale_order.csv",
+            }
+        )
+
+        # Search Model id to pass as an argument in `update error_log`
+        ir_model = self.env["ir.model"].search([("name", "=", "sale.order")])
+
+        # Test the method `update_error_log` for if condition.
+        import_log_id = self.import_sale_obj._update_error_log(
+            error_log_id=False,
+            error_vals={
+                "error": "Data not found",
+                "error_name": "Test Product Does not found",
+            },
+            ir_attachment=ir_attachment,
+            model=ir_model,
+            row_no=2,
+            order_group_value=1,
+        )
+
+        # Search the error log which is created by `update_error_log`
+        error_log_id = self.env["error.log"].search(
+            [("model_id", "=", ir_model.id), ("state", "=", "failed")]
+        )
+
+        # Compare the values which generated by `update_error_log` method
+        self.assertEqual(
+            error_log_id.input_file, ir_attachment, "Check the uploaded attachment"
+        )
+        self.assertEqual(
+            error_log_id.import_user_id,
+            self.env.user,
+            "Check the environment user seems not different",
+        )
+
+        # Test the method `update_error_log` for `else` condition.
```

Comment:
> for `else` condition.

Not a good way to describe the condition as the code may change. Please state the condition in a concrete manner.